src/common/utils/async-result.ts

```diff
@@ -3,5 +3,9 @@
  * Licensed under MIT License. See LICENSE in root directory for more information.
  */
 export type AsyncResult<Response, Error = string> =
-  | { callWasSuccessful: true; response: Response }
+  | (
+    Response extends void
+    ? { callWasSuccessful: true; response?: undefined }
+    : { callWasSuccessful: true; response: Response }
+  )
```

Comment:
This doesn't seem right. If the expected success response is undefined, then it should be used as a `Response` parameter. Now this means that we have to always make sure that `response` exists, which isn't fine.

Reply:
Read: https://github.com/lensapp/lens/pull/6096/files#r1000512666

Reply:
> This doesn't seem right. If the expected success response is undefined, then it should be used as a Response parameter. Now this means that we have to always make sure that response exists, which isn't fine.


Misread the code, it seems fine.